Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -497,7 +500,60 @@ fileprivate extension Array<AnnotatedTestItem> {
   }
 }
 
+fileprivate extension Array<TestItem> {
+  /// If multiple testItems share the same ID we add more context to make it unique.
+  /// Two tests can share the same ID when two swift testing tests accept
+  /// arguments of different types, i.e:
+  /// ```
+  /// @Test(arguments: [1,2,3]) func foo(_ x: Int) {}
+  /// @Test(arguments: ["a", "b", "c"]) func foo(_ x: String) {}
+  /// ```
+  ///
+  /// or when tests are in separate files but don't conflict because they are marked
+  /// private, i.e:
+  /// ```
+  /// File1.swift: @Test private func foo() {}
+  /// File2.swift: @Test private func foo() {}
+  /// ```
+  ///
+  /// If we encounter one of these cases, we need to deduplicate the ID
+  /// by appending `/filename:filename:lineNumber`.
+  func deduplicateIds() -> [TestItem] {
```

Comment:
Because this is not modifying the array in place, it should be named `deduplicatingIds` according to the [naming guidelines](https://www.swift.org/documentation/api-design-guidelines/#name-according-to-side-effects)